instance/instance.go

```diff
@@ -610,28 +623,70 @@ func writeBody(w http.ResponseWriter, b []byte) {
 // PublishCheck Checks if an instance has been published
 type PublishCheck struct {
 	Datastore store.Storer
+	Auditor   audit.AuditorService
 }
 
 // Check wraps a HTTP handle. Checks that the state is not published
-func (d *PublishCheck) Check(handle func(http.ResponseWriter, *http.Request)) http.HandlerFunc {
+func (d *PublishCheck) Check(handle func(http.ResponseWriter, *http.Request), action string) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-
+		ctx := r.Context()
 		vars := mux.Vars(r)
-		id := vars["id"]
-		instance, err := d.Datastore.GetInstance(id)
-		if err != nil {
-			log.Error(err, nil)
-			handleErrorType(err, w)
+		instanceID := vars["id"]
+		logData := log.Data{"instance_id": instanceID}
+		auditParams := common.Params{"instance_id": instanceID}
+
+		if err := d.Auditor.Record(ctx, action, audit.Attempted, auditParams); err != nil {
+			audit.LogActionFailure(ctx, action, audit.Attempted, err, logData)
+			handleErr(ctx, errs.ErrAuditActionAttemptedFailure, 0, w, logData)
 			return
 		}
 
-		if instance.State == models.PublishedState {
-			err = errors.New("unable to update instance as it has been published")
-			log.Error(err, log.Data{"instance": instance})
-			http.Error(w, err.Error(), http.StatusForbidden)
+		statusCode, err := d.checkState(w, instanceID)
+		if err != nil {
+			log.ErrorCtx(ctx, errors.WithMessage(err, "errored whilst checking instance state"), logData)
+			if auditErr := d.Auditor.Record(ctx, action, audit.Unsuccessful, auditParams); auditErr != nil {
+				audit.LogActionFailure(ctx, action, audit.Unsuccessful, auditErr, logData)
+				handleErr(ctx, errs.ErrAuditActionAttemptedFailure, 0, w, logData)
+				return
+			}
+
+			handleErr(ctx, err, statusCode, w, logData)
 			return
 		}
 
 		handle(w, r)
 	})
 }
+
+func (d *PublishCheck) checkState(w http.ResponseWriter, instanceID string) (int, error) {
+	instance, err := d.Datastore.GetInstance(instanceID)
+	if err != nil {
+		return 0, err
+	}
+
+	if instance.State == models.PublishedState {
+		err = errors.New("unable to update instance as it has been published")
+		return 403, err
+	}
+
+	return 0, nil
+}
+
+func handleErr(ctx context.Context, err error, status int, w http.ResponseWriter, data log.Data) {
```

Comment:
The int parameter is only required because of the 403 case mentioned above. If you create a new error type to handle the 403 scenario, then this function can be simplified quite lot. Something like this: (not tested)
```
func handleErr(ctx context.Context, err error, w http.ResponseWriter, data log.Data) {
	if data == nil {
		data = log.Data{}
	}

	status := http.StatusInternalServerError
	
	switch {
	case err == errs.ErrDatasetNotFound || err == errs.ErrInstanceNotFound || err == errs.ErrDimensionNodeNotFound:
		status = http.StatusNotFound
	case err == errs.ErrForbidden
		status = http.StatusForbidden
	}

	data["responseStatus"] = status
	audit.LogError(ctx, errors.WithMessage(err, "request unsuccessful"), data)
	http.Error(w, err.Error(), status)
}
```